index.js

```diff
@@ -53,170 +43,75 @@ async function main () {
-  core = typeof core === 'string' ? store.get(HypercoreId.decode(core)) : core
-
-  const bee = new Hyperbee(core)
-  const watcher = watch ? bee.watch() : null
-  await bee.ready()
-
-  if (track) tracker.add('bee', bee, { speedometer: bee.core })
-
-  await downloadCore(core, announce, { track: false, parent: bee })
-
-  return { bee, watcher }
-}
-
-async function downloadDrive (key, announce) {
-  const drive = new Hyperdrive(store, HypercoreId.decode(key))
-  await drive.ready()
-
-  const info = tracker.add('drive', drive, { speedometer: drive.core })
-  if (drive.blobs) Tracker.speedometer(drive.blobs.core, info)
-  else drive.on('blobs', blobs => Tracker.speedometer(blobs.core, info))
-
-  downloadBee(drive.core, announce, { track: false })
-  if (drive.blobs) downloadCore(drive.blobs.core, false, { track: false, parent: drive })
-  else drive.on('blobs', blobs => downloadCore(blobs.core, false, { track: false, parent: drive }))
-}
-
-async function downloadSeeder (key, { drives }) {
-  const publicKey = HypercoreId.decode(key)
-  const id = HypercoreId.encode(publicKey)
-  const keyPair = await store.createKeyPair('simple-seeder-swarm@' + id)
-  const sw = new Seeders(publicKey, { dht: swarm.dht, keyPair })
-
-  tracker.add('seeder', sw)
-
-  sw.join()
-  sw.on('connection', onsocket)
-  sw.on('update', function (record) {
-    tracking.notifs[id] = record
-  })
-
-  const unregister = goodbye(() => sw.destroy())
-  sw.on('close', () => unregister())
-
-  if (!drives.includes(key)) {
-    await downloadDrive(key, false)
-  }
-}
-
-async function downloadLists (core, announce) {
-  const { bee, watcher } = await downloadBee(core, announce, { track: false, watch: !argv['dry-run'] })
-
-  tracker.add('list', bee, { speedometer: bee.core, watcher })
-
-  return bee
-}
-
-async function watcher () {
-  const lists = tracker.getByType('list')
-  if (!lists.length) return
-
-  for (const { list, watcher } of lists) {
-    if (!watcher) continue
-    watching(list, watcher) // TODO: safety catch
+  const intervalId = setInterval(ui, argv.i || 5000)
+  goodbye(() => clearInterval(intervalId)) // Explicit cleanup
+  ui()
+
+  const resource = tracker.find(r => r.type === 'list')
+  if (resource && !argv['dry-run']) {
+    const list = resource.instance
+    // TODO: catch errors somewhere
+    const debounced = debounceify(update.bind(null, tracker, list))
+    list.core.on('append', debounced)
+    await debounced()
   }
 }
 
-async function watching (bee, watcher) {
-  /* for await (const [current] of watcher) {
-    for await (const entry of current.createReadStream()) {
-      const resources = tracker.getByType(entry.value.type)
-      const resource = resources.find(filter(entry))
-
-      if (!resource) {
-
-      }
-
-      if (resource) {
-        // TODO: skip same
-      }
+async function update (tracker, list) {
+  for (const [, info] of tracker.resources) {
```

Comment:
expose a .values() helper for this that does tracker.resources.values()

Reply:
Instead of the helper just using `tracker.resources.values()` directly now, or you actually prefer the `values` method?

Reply:
would say that resources is internal and an outsider shouldn't iterate it